OPENCAST-2359: Pre-populated lecture recording feedback form

npmali opened the pull request.

npmali committed:
OPENCAST-2359: Pre-populated lecture recording feedback form

smarquard commented:
As I understand it, you're getting the series id from listing the mediapackages. But it's possible there's nothing yet scheduled or published in the series. Also if there are recordings, we don't know that the feedback relates to any specific recording.

So the safest thing to do is use the series id that's passed in the LTI custom parameters.

npmali committed:
OPENCAST-3259: Updated series ID and course title

Changed the mediapackage series ID to the series ID passed in the LTI Custom parameters and used it to get the course title using the series metadata API

npmali commented:
Thanks Stephen, I've updated the script.

smarquard commented:
Everything that relates to an individual mediapackage should go - if there are multiple mediapackages, we don't know which one the feedback relates to (if it's about any - could be about missing recording), so it would be misleading to include mediapackage id or date.

TurRil commented on modules/lti/src/main/resources/tools/courses/js/lti.recordings.js:
```diff
@@ -75,6 +75,54 @@ function xhr(params, cb, fail, always) {
+  if (tempVars.mediaPackage_seriesid) {
+    parts.push('entry.1638000924=' + tempVars.mediaPackage_seriesid);
+  }
+  if (tempVars.mediaPackage_date) {
```
The mediapackage date is not really relevant because this feedback is essentially for the entire series and not for s specific recording so the date field will be incorrect. So this should not be filled in or sent to the google form.

npmali replied:
Changes pushed

npmali committed:
OPENCAST-2359: Unpopulated date field
